### PR TITLE
data: fix out-of-spec signatures

### PR DESCRIPTION
**Pull request by lourkeur**
as mentioned on Slack, I messed up in #1000 an inserted some extra embellishment, shall we say, to the signature payload.

### CODE REVIEW

**Comment by MaximeZmt** [comment]
Hey, after having tested one of your json, I still cannot verify it on front-end 2 😅 
As I cannot put custom private key I had to regenerate some keypair for the example.
Here are some data that I can sign and then validate, can you try them on your side:

PostTransactionCoin{ 
transaction_id=01N1Y8twdu7wpdz5HLnkIeQSeuKpkNcQHeKF7XabLYU=, 
transaction=
Transaction{version=1, 
inputs=[input{tx_out_hash='AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=', tx_out_index=0, script=script{type='P2PKH', pubkey='PublicKey(O2vB2TXoiyMzo70igu1syG_Op3v0ASP3pZ1hJwlQL2U=)', sig='Signature(434llLRpyCLWY2RFqURBViPnPHDoSdL_V6K_IFhPEDOaZjt3tNgRc5DfqGtvBzksX_Has6Pt4ygKmHprkJT2CQ==)'}}], 
outputs=[output{value=32, script=script{type='P2PKH', pubkey_hash='2bDfYcf0s1-CxqlGR3Zew2ZzNBg='}}], 
lock_time=0}
}

PoPToken{privateKey=PlainPrivateKey@70c34d09, publicKey=PublicKey(O2vB2TXoiyMzo70igu1syG_Op3v0ASP3pZ1hJwlQL2U=)}  /*SENDER + Sign*/

PoPToken{privateKey=PlainPrivateKey@2a36a815, publicKey=PublicKey(EeSjLu9YjOFBYgZU9Uq3zr48gLbqF2SXn78SIyaBeow=)} /*Reciever*/

TextToBeSigned: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=032P2PKH2bDfYcf0s1-CxqlGR3Zew2ZzNBg=

Encoded Base64 bef sign: QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQT0wMzJQMlBLSDJiRGZZY2YwczEtQ3hxbEdSM1pldzJaek5CZz0=


Which I can verify with the following method:
transactionModel.getInputs().get(0).getScript().getPubkey().verify(sig, data)
-> takes the public key of input, then verify the Signature of that given input and the data above encoded as base64

**Comment by MaximeZmt** [comment]
> Hey, after having tested one of your json, I still cannot verify it on front-end 2 😅 As I cannot put custom private key I had to regenerate some keypair for the example. Here are some data that I can sign and then validate, can you try them on your side:
> 
> PostTransactionCoin{ transaction_id=01N1Y8twdu7wpdz5HLnkIeQSeuKpkNcQHeKF7XabLYU=, transaction= Transaction{version=1, inputs=[input{tx_out_hash='AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=', tx_out_index=0, script=script{type='P2PKH', pubkey='PublicKey(O2vB2TXoiyMzo70igu1syG_Op3v0ASP3pZ1hJwlQL2U=)', sig='Signature(434llLRpyCLWY2RFqURBViPnPHDoSdL_V6K_IFhPEDOaZjt3tNgRc5DfqGtvBzksX_Has6Pt4ygKmHprkJT2CQ==)'}}], outputs=[output{value=32, script=script{type='P2PKH', pubkey_hash='2bDfYcf0s1-CxqlGR3Zew2ZzNBg='}}], lock_time=0} }
> 
> PoPToken{privateKey=PlainPrivateKey@70c34d09, publicKey=PublicKey(O2vB2TXoiyMzo70igu1syG_Op3v0ASP3pZ1hJwlQL2U=)} /_SENDER + Sign_/
> 
> PoPToken{privateKey=PlainPrivateKey@2a36a815, publicKey=PublicKey(EeSjLu9YjOFBYgZU9Uq3zr48gLbqF2SXn78SIyaBeow=)} /_Reciever_/
> 
> TextToBeSigned: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=032P2PKH2bDfYcf0s1-CxqlGR3Zew2ZzNBg=
> 
> Encoded Base64 bef sign: QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQT0wMzJQMlBLSDJiRGZZY2YwczEtQ3hxbEdSM1pldzJaek5CZz0=
> 
> Which I can verify with the following method: transactionModel.getInputs().get(0).getScript().getPubkey().verify(sig, data) -> takes the public key of input, then verify the Signature of that given input and the data above encoded as base64

it has been generated with the following pr commit [c003da0](https://github.com/dedis/popstellar/pull/1097/commits/c003da0c74337965422935d067b1679597774ff6)

(The transaction id has not been regenerated so it might be wrong, but it is not the problem here :) )

**Comment by lourkeur** [comment]
@MaximeZmt I tested this signature on the Scala back-end, it looks like it's valid 683b2da8787121a0f56c28e0c9b60fa35703ae41

**Review by MaximeZmt — APPROVED** [review]
👍 Tested by hand on fe-2 looks perfect 😃  !

File: be2-scala/src/test/scala/ch/epfl/pop/model/objects/TransactionSuite.scala [hunk 1]
```diff
@@ -43,4 +43,11 @@ class TransactionSuite extends FunSuite with Matchers {
     val signedTransaction = transaction.sign(Seq(keypairs(1).keyPair))
     signedTransaction.checkSignatures() shouldEqual true
   }
+
+  test("signature is performed over the correct data") {
```
Comment (lourkeur) [inline be2-scala/src/test/scala/ch/epfl/pop/model/objects/TransactionSuite.scala:47]: adding this as penance because I should have done it to begin with

**Review by haoqianzhang — APPROVED** [review]

**Pull request merged**